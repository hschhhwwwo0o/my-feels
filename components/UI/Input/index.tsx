import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

interface IInput {
  type?: "text" | "password";
  placeholder?: string;
  value?: string;
  setValue?: Dispatch<SetStateAction<string>>;
}

const Input: FunctionComponent<IInput> = ({
  type = "text",
  placeholder = "Enter text...",
  value = "",
  setValue = () => {},
}) => {
  return (
    <div>
      <input
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={e => {
          setValue(e.target.value);
        }}
        className="bg-[#FAFAFA] border-2 w-full border-[#EAEAEA] outline-none px-8 py-5 rounded-md mulish font-semibold text-[#242424] dark:text-[#8D8D8D] transition-all duration-1000 focus:border-[#dadada] dark:focus:border-[#444444] placeholder-[#D3D3D3] dark:placeholder-[#8D8D8D] dark:bg-[#151515] dark:border-[#2D2D2D]"
      />
    </div>
  );
};

export default Input;
