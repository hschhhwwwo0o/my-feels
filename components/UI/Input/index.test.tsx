import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Input from "./index";

test("Test Input component", () => {
  render(
    <span data-testid="block">
      <Input value="" setValue={() => ""} />
    </span>
  );
  const el: HTMLElement = screen.getByTestId("block");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
