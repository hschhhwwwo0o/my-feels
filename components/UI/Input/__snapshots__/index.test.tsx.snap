// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component 1`] = `
<span
  data-testid="block"
>
  <div>
    <input
      class="bg-[#FAFAFA] border-2 w-full border-[#EAEAEA] outline-none px-8 py-5 rounded-md mulish font-semibold text-[#242424] dark:text-[#8D8D8D] transition-all duration-1000 focus:border-[#dadada] dark:focus:border-[#444444] placeholder-[#D3D3D3] dark:placeholder-[#8D8D8D] dark:bg-[#151515] dark:border-[#2D2D2D]"
      placeholder="Enter text..."
      type="text"
      value=""
    />
  </div>
</span>
`;
